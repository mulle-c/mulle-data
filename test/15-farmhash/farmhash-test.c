#include <mulle-data/mulle-data.h>
#include <stdio.h>


int   main( void)
{
   uintptr_t  hash;

   hash = mulle_hash( NULL, 0);
   printf( "0x%016tx\n", hash);

   hash = mulle_hash( "VfL Bochum 1848", 15);
   printf( "data: %#016tx\n", hash);

   hash = mulle_integer_hash( 1848);
   printf( "integer: %#016tx\n", hash);

   hash = mulle_long_long_hash( 1848);
   printf( "long long %#016tx\n", hash);

   hash = mulle_float_hash( 1848.0);
   printf( "float %#016tx\n", hash);

   hash = mulle_double_hash( 1848.0);
   printf( "double: %#016tx\n", hash);

   hash = mulle_pointer_hash( (void *) 1848);
   printf( "pointer: %#016tx\n", hash);

   // leftovers
   hash = _mulle_hash_32( "VfL Bochum 1848", 15);
   printf( "32: %#016tx\n", hash);

   hash = _mulle_hash_64( "VfL Bochum 1848", 15);
   printf( "64: %#016tx\n", hash);

   hash = _mulle_hash_chained_32( "VfL Bochum 1848", 15, 0);
   printf( "chain32: %#016tx\n", hash);

   hash = _mulle_hash_chained_64( "VfL Bochum 1848", 15, 0);
   printf( "hash32: %#016tx\n", hash);

   return( 0);
}