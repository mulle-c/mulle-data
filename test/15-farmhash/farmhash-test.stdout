0x0000000000000000
data: 0x2450f7287f11f328
integer: 0x23cc2a942c86a786
long long 0x23cc2a942c86a786
float 0xc2d1a2481b3ff300
double: 0xc2d1a2481b3ff300
pointer: 0x23cc2a942c86a786
32: 0x000000ced51853
64: 0x2450f7287f11f328
chain32: 0x000000ced51853
hash32: 0xbd1a7a8aec809763
